plugins/chat/lib/chat_channel_hashtag_data_source.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class Chat::ChatChannelHashtagDataSource
+  def self.icon
```

Comment:
Is there any interest to have this as a method here ?

Reply:
I am not sure what you mean? I need a way to easily query this for each data source because I pass the icons to the markdown flow via `hashtagIcons`

Reply:
You query this on `HashtagItem` after transforming with `to_h`. But this `def self.icon` is creating a public method on `Chat::ChatChannelHashtagDataSource` which could just be a constant

Reply:
I think a method makes more sense since we want all these classes to be polymorphic. Why class methods though? 😛

Reply:
I think a method makes more sense since we want all these classes to be polymorphic. Why class methods though? 😛

Reply:
See this method, we need a way to easily pass this down to the app via Site for our markdown flow:

https://github.com/discourse/discourse/blob/56d275549b28e1198e3e022bd30220322ab39410/app/services/hashtag_autocomplete_service.rb#L30-L32

Here is the relevant place I use it in the markdown rule:

https://github.com/discourse/discourse/blob/56d275549b28e1198e3e022bd30220322ab39410/app/assets/javascripts/pretty-text/engines/discourse-markdown/hashtag-autocomplete.js#L95-L108